src/ClientServices/include/ClientServices/ProcessManager.h

```diff
@@ -18,34 +18,20 @@
 
 #include "GrpcProtos/module.pb.h"
 #include "GrpcProtos/process.pb.h"
+#include "GrpcProtos/services.pb.h"
 #include "GrpcProtos/symbol.pb.h"
 #include "OrbitBase/Result.h"
 #include "absl/synchronization/mutex.h"
 
 namespace orbit_client_services {
 
-// This class is responsible for maintaining
-// process list. It periodically updates it
-// and calls callback to notify listeners when
-// the list is updated.
-//
-// Usage example:
-//
-// auto manager = ProcessListManager::Create(...);
-// auto callback = [&](const std::vector<ProcessInfo>& process_list) {
-//   // Update process list in UI
-// }
-// manager.SetProcessListUpdateListener(callback);
-//
-// To orderly shutdown the manager use following:
-//
-// manager.Shutdown();
-//
+// ProcessManager is used to access or launch processes on the target machine through OrbitService.
```

Comment:
The existing comment was outdated. It was describing the ProcessManager as a ProcessListManager only responsible for periodically providing a list of processes. I think the current change leaves ProcessManager in a better state actually. Concerning the fact that we are adding process launching functionality, I think it fits well into a process manager. We could split functionality into two managers, but then should we also split up the existing different functionalities (LoadModuleList, LoadProcessMemory, FindDebugInfoFile...)? It also maps well to the ProcessClient, which the ProcessManager already owns. I prefer keeping as is.

Reply:
I have to disagree here. First of all, "process manager" is just a name, but that doesn't mean that everything that is process-related should use it. What I'm looking at is responsibilities of the class, and if they no longer match the names, we can rename. Yes, it's true that these responsibilities are already larger than originally intended, but let's not make things worse.
In particular, this is not just a new method, but a completely new set of methods that are connected together and are related to the same functionality, and that are completely separate from the existing functionalities of the class. And as wrote in the other comment, this is shown by the fact that you are adding a new mutex and a new field that are strictly only for the new methods. But the fact alone that those new methods are grouped together tells you the same. This for me makes it super clear that there should be a new class. 

Honestly, the same applies to the service itself. I think we should just have a new service, something like `LauncherService` or `ProcessLauncherService`. Otherwise we are just making the `ProcessService` the catch-all service because we don't want to add new one.